engine/src/main/java/com/datatorrent/stram/LaunchContainerRunnable.java

```diff
@@ -230,6 +230,12 @@ public void run()
       }
     }
 
+    String loggerAppender = dag.getValue(LogicalPlan.LOGGER_APPENDER);
+    if (loggerAppender != null) {
+      vargs.add(String.format("-D%s=\"%s\"", LogicalPlan.LOGGER_APPENDER.getName(), loggerAppender));
+      vargs.add(String.format("-D%s=\"%s\"", LogicalPlan.APPLICATION_NAME.getName(), dag.getValue(LogicalPlan.APPLICATION_NAME)));
```

Comment:
The application name like the other application properties is a property of DAG. And all application properties are available after reading of the logical/physical plan. The value of the container id property is already passed via the environment variable CONTAINER_ID.